attest/untrusted/src/sim.rs

```diff
@@ -53,37 +53,51 @@ impl SimQuotingEnclave {
     }
 
     pub fn collateral<Q: AsRef<[u8]>>(_quote: &Quote3<Q>) -> Collateral {
-        let mut tcb_info = SIM_TCB_INFO.to_owned();
-        let mut qe_identity = SIM_QE_IDENTITY.to_owned();
-
-        let mut crl = SIM_CRL.to_vec();
-        crl.push(0);
-
-        // Normally the PCK chain would not be the same one used on TCB info and the QE
-        // identity, but they share the same root so it works for this sim
-        // implementation.
-        let mut pem_chain = IAS_SIM_SIGNING_CHAIN.as_bytes().to_vec();
-
-        let mut sgx_collateral = version_3_1_empty_collateral();
-        sgx_collateral.root_ca_crl = crl.as_mut_ptr() as *mut core::ffi::c_char;
-        sgx_collateral.root_ca_crl_size = crl.len() as u32;
-        sgx_collateral.pck_crl_issuer_chain = pem_chain.as_mut_ptr() as *mut core::ffi::c_char;
-        sgx_collateral.pck_crl_issuer_chain_size = pem_chain.len() as u32;
-        sgx_collateral.pck_crl = crl.as_mut_ptr() as *mut core::ffi::c_char;
-        sgx_collateral.pck_crl_size = crl.len() as u32;
-        sgx_collateral.tcb_info_issuer_chain = pem_chain.as_mut_ptr() as *mut core::ffi::c_char;
-        sgx_collateral.tcb_info_issuer_chain_size = pem_chain.len() as u32;
-        sgx_collateral.tcb_info = tcb_info.as_mut_ptr() as *mut core::ffi::c_char;
-        sgx_collateral.tcb_info_size = tcb_info.len() as u32;
-        sgx_collateral.qe_identity_issuer_chain = pem_chain.as_mut_ptr() as *mut core::ffi::c_char;
-        sgx_collateral.qe_identity_issuer_chain_size = pem_chain.len() as u32;
-        sgx_collateral.qe_identity = qe_identity.as_mut_ptr() as *mut core::ffi::c_char;
-        sgx_collateral.qe_identity_size = qe_identity.len() as u32;
-
-        Collateral::try_from(&sgx_collateral).expect("Failed to convert collateral")
+        collateral_from_from_raw_data(
+            SIM_TCB_INFO,
+            SIM_QE_IDENTITY,
+            SIM_CRL,
+            IAS_SIM_SIGNING_CHAIN,
+        )
     }
 }
 
+fn collateral_from_from_raw_data(
```

Comment:
Up for naming suggestion here, I wanted to make it easier to construct one off `Collateral`s for the tests below